src/step_definitions/performance.js

```diff
@@ -43,7 +45,36 @@ defineSupportCode(function ({ When }) {
     });
 
     return browser.driver.wait(() => {
-      return proxyDone;
+      return (
+        this.performanceReportFile = uniqueFileName,
+        proxyDone
+      )
     });
   });
+
+  Then(/^the requests should take a maximum of "([^"]*)" milliseconds$/, function (maxTiming) {
+    const performanceReport = JSON.parse(fs.readFileSync(`reports/performance/${this.performanceReportFile}`, 'utf8'));
+    const mappedRequests = performanceReport.log.entries.map(item => ({
+      ttfb: item.timings.wait,
+      url: item.request.url
+    }));
```

Comment:
For the sake of clarity, those lines above could be put to some module related to performance as a function like `getPerformanceReport(filename)`.
This way you'll have cleaner code here and you'll be able to read this file in other places without duplicating the code.

Going further: you could think of creating a function `getPerformanceReport(filename, maxTiming)`, which will return only the requests that exceeded the given timing - even more cleaner code.

Reply:
I would rather create a PerformanceReportParse, where in constructor we will require a filename and timing and the only method will be a method "parse".

As a result we will get a list of requests or even better an object indicating if there are slow requests or not. This could be used to reject/resolve a step.

Reply:
The code has been moved to a `helper`.
`parse()` method is used in step.